tests/test_fmi3.py

```diff
@@ -580,35 +596,109 @@ def test_get_input_list(self):
+            ("get_int8", [11, 12], np.array([0, 0], dtype=np.int8), np.int8),
+            ("get_uint64", [25, 26], np.array([0, 0], dtype=np.uint64), np.uint64),
+            ("get_uint32", [21, 22], np.array([0, 0], dtype=np.uint32), np.uint32),
+            ("get_uint16", [17, 18], np.array([0, 0], dtype=np.uint16), np.uint16),
+            ("get_uint8", [13, 14], np.array([0, 0], dtype=np.uint8), np.uint8),
+            ("get_boolean", [27, 28], np.array([False, False], dtype=np.bool_), np.bool_),
+            ("get_enum", [32, 33], np.array([1, 1], dtype=np.int64), np.int64),
+        ]
+    )
+    def test_getX(self, function_name, valuerefs, expected_result, expected_dtype):
+        """Test the various get_TYPE([<valueref>]) functions."""
         fmu_path = FMI3_REF_FMU_PATH / "Feedthrough.fmu"
         fmu = load_fmu(fmu_path)
-        int32_vals = fmu.get_int32([19, 20])
+        res = getattr(fmu, function_name)(valuerefs) # fmu.<function_name>(valuerefs)
 
-        np.testing.assert_equal(np.array([0, 0], dtype=np.int32), int32_vals)
+        np.testing.assert_equal(res, expected_result)
+        assert res.dtype == expected_dtype
+
+    def test_get_string(self):
+        """Test the get_string function."""
+        fmu_path = FMI3_REF_FMU_PATH / "Feedthrough.fmu"
+        fmu = load_fmu(fmu_path)
+        res = fmu.get_string([29])
 
-        # Since above doesnt verify the data type, also check below separately
-        assert int32_vals.dtype == np.int32
+        np.testing.assert_equal(res, ["Set me!"])
+        assert type(res) == list
 
-    def test_get_int64(self):
-        """ Test get int64. """
+    @pytest.mark.parametrize("variable_name, value, expected_dtype",
+        [
+            ("Float64_continuous_input", 3.14, np.double),
+            ("Float32_continuous_input", np.float32(3.14), np.float32),
+            ("Int64_input", 9_223_372_036_854_775_806, np.int64),
+            ("Int32_input", 2_147_483_647, np.int32),
+            ("Int16_input", 32_766, np.int16),
+            ("Int8_input", 126, np.int8),
+            ("UInt64_input", 18_446_744_073_709_551_615, np.uint64),
+            ("UInt32_input", 4_294_967_294, np.uint32),
+            ("UInt16_input", 65_534, np.uint16),
+            ("UInt8_input", 254, np.uint8),
+            ("Boolean_input", True, np.bool_),
+            ("Enumeration_input", 2, np.int64),
+        ]
+    )
+    def test_set_get(self, variable_name, value, expected_dtype):
+        """Test getting and setting variables of various types."""
         fmu_path = FMI3_REF_FMU_PATH / "Feedthrough.fmu"
         fmu = load_fmu(fmu_path)
-        int64_vals = fmu.get_int64([23, 24])
+        fmu.set(variable_name, value)
+        res = fmu.get(variable_name)
+        assert res.dtype == expected_dtype
+        assert res[0] == value
 
-        np.testing.assert_equal(np.array([0, 0]), int64_vals)
+    @pytest.mark.skipif(NUMPY_MAJOR_VERSION > 1, reason = "Error for numpy>=2")
```

Comment:
While the test is effectively somewhat duplicated, trying to follow the guideline of "don't use `ifs` depending on the parameterization in unit tests".